### PR TITLE
[DataGrid] Add missing TablePagination localizations

### LINKED ISSUE #1107: [DataGrid] Pagination localization
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Currently of the footer of the table look like this while using pagination:
108724352-83c68f80-7525-11eb-84ff-af8a01f825b6

The word "of" is somehow not changeable and it affects the localization of the DataGrid when implementing it on another language.

## Context 🔦

I am trying to implement a DataGrid with Pagination on German. Everything seems fine until this small issue.

### PR DESCRIPTION
**Pull request by DanailH**
Fixes #1107 

Some context about the discussion:

> The points I wonder about:
> 1. Do we need to duplicate the translation keys for the components? Do developers need to translate a <TablePagination> differently in the context of a data grid? (DataGrid.localeText.footerPaginationRowsPerPage vs. TablePagination.labelRowsPerPage)
> 2. Should we duplicate (seems we need the same translations) the maintenance of the translated locales between the two components?
> 3. Should the data grid ignore the translation of the core components and ask developers to configure them independently? Basically, asking to do https://material-ui.com/guides/localization/#locale-text. And in the future, if we support importing individual translations, to cherry-pick what they need.
> 4. Should the data grid work in isolation, importing its translation from the core package, avoiding the need for 3?

### CODE REVIEW

**Comment by oliviertassinari** [comment]
@DanailH I have fixed the build on HEAD. You can rebase.

**Comment by DanailH** [comment]
@oliviertassinari I adjusted the translation. I removed the one that we had for the pagination and am now pulling the ones from the core to basically extend them with the ones specific for the grid.

File: packages/grid/_modules_/grid/utils/getGridLocalization.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import * as locales from '@material-ui/core/locale';
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/utils/getGridLocalization.ts:1]: This breaks tree-shaking, we need to import each locale in each translation file

**Review by oliviertassinari — CHANGES_REQUESTED** [review]

File: packages/grid/_modules_/grid/utils/getGridLocalization.ts (hunk 1 above)
Comment (DanailH) [reply packages/grid/_modules_/grid/utils/getGridLocalization.ts:1]: I think it needs to be here, I can import only the ones we need but this is where the object is built.

Comment (oliviertassinari) [reply packages/grid/_modules_/grid/utils/getGridLocalization.ts:1]: Simply do:

```jsx
import { bgBG }  from '@material-ui/core/locale';
```

in the `bgBG` locale file and so one for each locale;

Comment (DanailH) [reply packages/grid/_modules_/grid/utils/getGridLocalization.ts]: Updated

**Review by oliviertassinari — APPROVED** [review]

**Comment by quocanh176** [comment]
Thanks a lot for the quick fixing!

**Comment by MoiseScalzo** [comment]
Same here, could not find a way to localize pagination strings :(

**Comment by farrellp43** [comment]
> Thanks a lot for the quick fixing! Just a quick question though. I took a look at the file "localeTextConstants.ts" from the latest commit and i still see no property to localize the word "of" for the pagination footer. Am i missing something?
> 
> @oliviertassinari @DanailH could anyone help me on my question?

Still waiting on how to localize the pagination

**Pull request merged**